src/GovUk.Education.ExploreEducationStatistics.Admin/Services/EmbedBlockService.cs

```diff
@@ -0,0 +1,157 @@
+                    });
+            });
+    }
+
+    public async Task<Either<ActionResult, EmbedBlockLinkViewModel>> Update(Guid releaseId,
+        Guid contentBlockId,
+        EmbedBlockUpdateRequest request)
+    {
+        return await _persistenceHelper
+            .CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async release =>
+            {
+                return await _persistenceHelper
+                    .CheckEntityExists<EmbedBlockLink>(contentBlockId, query =>
+                        query.Include(cb => cb.EmbedBlock))
+                    .OnSuccess<ActionResult, EmbedBlockLink, EmbedBlockLinkViewModel>(async contentBlock =>
+                    {
+                        if (!_contentDbContext.ReleaseContentSections
+                            .Any(rcs => rcs.ReleaseId == release.Id
+                            && rcs.ContentSectionId == contentBlock.ContentSectionId))
+                        {
+                            return ValidationUtils.ValidationActionResult(
+                                ValidationErrorMessages.ContentSectionNotAttachedToRelease);
+                        }
+
+                        var embedBlock = contentBlock.EmbedBlock;
+
+                        embedBlock.Title = request.Title;
+                        embedBlock.Url = request.Url;
+
+                        _contentDbContext.Update(embedBlock);
+                        await _contentDbContext.SaveChangesAsync();
+
+                        return _mapper.Map<EmbedBlockLinkViewModel>(contentBlock);
+                    });
+            });
+    }
+
+    public async Task<Either<ActionResult, Unit>> Delete(Guid releaseId, Guid contentBlockId)
+    {
+        return await _persistenceHelper
+            .CheckEntityExists<Release>(releaseId)
+            .OnSuccessDo(_userService.CheckCanUpdateRelease)
+            .OnSuccessVoid(async release =>
+            {
+                return await _persistenceHelper
+                    .CheckEntityExists<EmbedBlockLink>(contentBlockId, query
+                        => query.Include(cb => cb.EmbedBlock))
+                    .OnSuccess<ActionResult, EmbedBlockLink, Unit>(async contentBlock =>
+                    {
+                        if (!_contentDbContext.ReleaseContentSections
+                            .Any(rcs => rcs.ReleaseId == release.Id
+                            && rcs.ContentSectionId == contentBlock.ContentSectionId))
+                        {
+                            return ValidationUtils.ValidationActionResult(
+                                ValidationErrorMessages.ContentBlockNotAttachedToRelease);
+                        }
+
+                        await _contentBlockService.DeleteContentBlockAndReorder(contentBlock.Id);
+                        await _contentDbContext.SaveChangesAsync();
+
+                        return Unit.Instance;
```

Comment:
Shouldn't need to return anything if we use `OnSuccessVoid`

Reply:
If I turn it into an `OnSuccessVoid` it complains because the same `OnSuccess` also returns a `ValidationErrorMessage` - I could split these particular lines into their own `OnSuccessVoid`, but that would mean that the OnSuccess for L140-148 would need to return a `Unit.Instance` anyway.

Reply:
Ah yeh, of course, d'oh!